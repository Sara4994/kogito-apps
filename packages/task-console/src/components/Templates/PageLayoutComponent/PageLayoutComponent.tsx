import React from 'react';
import {
  Nav,
  NavList,
  NavItem,
  PageSection,
  Title
} from '@patternfly/react-core';
import { PageLayout } from '@kogito-apps/common/src/components';
import { withRouter } from 'react-router-dom';
import taskConsoleLogo from '../../../static/taskConsoleLogo.svg';

const PageLayoutComponent = props => {
  const PageNav = (
    <Nav aria-label="Nav" theme="dark" css="">
      <NavList>
        <NavItem>TestOption-1</NavItem>
        <NavItem isActive>TestOption-2</NavItem>
      </NavList>
    </Nav>
  );

  const BrandClick = () => {
    props.history.push('/');
  };

  return (
    <PageLayout
      PageNav={PageNav}
      BrandSrc={taskConsoleLogo}
      BrandAltText="Task Console Logo"
      BrandClick={BrandClick}
    >
      <PageSection variant="light">
        <Title headingLevel="h1" size="4xl">
          Task List
        </Title>
      </PageSection>{' '}
      {''}
    </PageLayout>
  );
};

export default withRouter(PageLayoutComponent);
