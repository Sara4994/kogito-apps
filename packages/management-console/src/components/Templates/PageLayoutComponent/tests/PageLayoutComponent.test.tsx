import React from 'react';
import { shallow } from 'enzyme';
import PageLayoutComponent from '../PageLayoutComponent';
import { MockedProvider } from '@apollo/react-testing';

const props: any = {
  location: {
    pathname: '/ProcessInstances'
  }
};

const mocks = [];

describe('PageLayoutComponent component tests', () => {
  it('snapshot testing', () => {
    const wrapper = shallow(
      <MockedProvider mocks={mocks}>
        <PageLayoutComponent {...props} />
      </MockedProvider>
    );
    expect(wrapper.find(PageLayoutComponent)).toMatchSnapshot();
  });
});
