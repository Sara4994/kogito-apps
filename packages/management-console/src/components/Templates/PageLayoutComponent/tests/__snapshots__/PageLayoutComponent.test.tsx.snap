// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageLayoutComponent component tests snapshot testing 1`] = `
<PageLayoutComponent
  location={
    Object {
      "pathname": "/ProcessInstances",
    }
  }
/>
`;
